tools/port-drayage-webservice/src/main/java/com/leidos/controller/InspectionController.java

```diff
@@ -0,0 +1,108 @@
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.RestController;
+
+/**
+ * {@link RestController} for all inspection/ REST endpoints. Implements
+ * {@link InspectionApi} generated by openapi-generator-maven-plugin (see
+ * pom.xml), which configures endpoint content type, response type and path.
+ * 
+ * @author Paul Bourelly
+ */
+@RestController
+public class InspectionController implements InspectionApi {
+
+    private static Logger logger = LoggerFactory.getLogger(InspectionController.class);
+
+    // Injected InspectionActions spring bean
+    @Autowired
+    private InspectionActions inspectionActions;
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public ResponseEntity<InspectionStatusList> inspectionPendingGet() {
+        return ResponseEntity.ok(inspectionActions.getPendingInspections());
+
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public ResponseEntity<InspectionStatus> inspectionActionIdGet(String actionId) {
+        InspectionStatus status = inspectionActions.getInspectionStatus(actionId);
+        logger.debug(String.format("Found inspection %s!", status));
+        if (status != null)
+            return ResponseEntity.ok(status);
+        return new ResponseEntity<InspectionStatus>(HttpStatus.BAD_REQUEST);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public ResponseEntity<Void> inspectionPost(InspectionRequest request) {
+        inspectionActions.requestInspectionAction(request);
+        return new ResponseEntity<>(HttpStatus.CREATED);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public ResponseEntity<Void> inspectionHoldingActionIdPost(String actionId) {
+        InspectionStatus cur = inspectionActions.getCurrentInspection();
+        if ( cur != null && cur.getActionId().equals(actionId)) {
+            inspectionActions.requestHolding();
+            return new ResponseEntity<>(HttpStatus.CREATED);
+        } else {
+            logger.warn(String.format("Action ID %s is not current inspection %s ", actionId,
+                    inspectionActions.getCurrentInspection().toString()));
+            return new ResponseEntity<>(HttpStatus.BAD_REQUEST);
+        }
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public ResponseEntity<Void> inspectionCompleteActionIdPost( String actionId ) {
+        InspectionStatus cur = inspectionActions.getCurrentInspection();
+        if (cur != null && cur.getActionId().equals(actionId)) {
+            inspectionActions.completeInspection();
+            return new ResponseEntity<>(HttpStatus.CREATED);
+        } else {
+            return new ResponseEntity<>(HttpStatus.BAD_REQUEST);
+        }
+
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public ResponseEntity<Void> inspectionHoldActionIdPost( String actionId) {
```

Comment:
After clicking the hold on UI, how to clear the proceed to hold, then proceed to the pending inspection list?
![image](https://user-images.githubusercontent.com/62157949/139700296-4805417b-8435-4189-ac4b-fff759000cc6.png)

Reply:
Currently there is no way to clear an in progress action. The way it would proceed to the next stage is the V2X-Hub Port Drayage Plugin would send a post to inspection/holding confirming the vehicle has arrived at the holding area.

Reply:
ok, I think the post endpoint is what I want to know. Thanks!